### DISCUSSION TITLE
Using Telegram TEST server

### DISCUSSION THREAD

**Discussion by OpperDev** [telegraf/telegraf#1847] [Q&A]
Its now possible to use Telegram on a Test server (https://core.telegram.org/bots/webapps#testing-web-apps) (Tap 10 times on settings etc... etc...)
Is it possible to use the api url "https://api.telegram.org/botTOKEN/test/" with Telegraf?

EDIT: Its working when im adding /test to my bot token, but its EXTREMELY slow

**Comment by MKRhere** [comment]
This is how you enable the test server:

```TS
new Telegraf(token, { telegram: { testEnv: true } });
```

> EXTREMELY slow

Telegram may simply be dedicating fewer resources to the test server, and it might be busy. Nothing Telegraf can do about it.
